File: python/templative/cli/distribute/simulatorCommands.py
from templative.lib.manage import instructionsLoader
from templative.lib.distribute.simulator import getSimulatorDirectory, writeSimulatorFile, convertToTabletopSimulator
import asyncclick as click

@click.command()
@click.option('-i', '--input', default=None, help='The directory of the produced game. Defaults to last produced directory.')
@click.option('-o', '--output', default=None, help='The Tabletop Simulator packages directory. Such as "~/Library/Documents/My Games/Tabletop Simulator" or "C:/Users/User/Documents/My Games/Tabletop Simulator"')
async def simulator(input, output):
    """Convert a produced game into a tabletop simulator game"""    
    if input is None:
        input = await instructionsLoader.getLastOutputFileDirectory()
    
    if input is None:
        raise Exception("Missing --input directory.")

    simulatorDirectory = await getSimulatorDirectory(output)
    if simulatorDirectory == None:
        print("Missing --output directory.")
        return
    await writeSimulatorFile(simulatorDirectory)

    return await convertToTabletopSimulator(input, simulatorDirectory)

File: src/main/templative/api/distribute/simulatorCommands.js
const { Command } = require('commander');
const path = require('path');
const fs = require('fs');
const { getLastOutputFileDirectory } = require('../../lib/manage/instructionsLoader');
const { lookForSimulatorFile, writeSimulatorFile, getSimulatorDirectory, convertToTabletopSimulator } = require('../../lib/distribute/simulator/simulator');

// Create the simulator command
const simulator = new Command('simulator')
  .description('Convert a produced game into a tabletop simulator game')
  .requiredOption('-i, --input <path>', 'The directory of the produced game. Defaults to last produced directory.')
  .requiredOption('-o, --output <path>', 'The Tabletop Simulator packages directory. Such as "~/Library/Documents/My Games/Tabletop Simulator" or "C:/Users/User/Documents/My Games/Tabletop Simulator"')
  .action(async (options) => {
    let inputPath = options.input;
    const outputPath = options.output;
    
    if (!inputPath) {
      inputPath = await getLastOutputFileDirectory();
    }
    
    if (!inputPath) {
      console.error("Missing --input directory.");
      process.exit(1);
    }
    console.log(inputPath);
    
    const simulatorDirectory = await getSimulatorDirectory(outputPath);
    if (!simulatorDirectory) {
      console.error("Missing --output directory.");
      return;
    }
    
    await writeSimulatorFile(simulatorDirectory);
    // const templativeToken = await getSessionToken();
    // if (!templativeToken) {
    //   return { success: false, error: 'You must be logged into Templative to create a Tabletop Simulator save.' };
    // }
    const templativeToken = null;
    return await convertToTabletopSimulator(inputPath, simulatorDirectory, templativeToken);
  });


module.exports = simulator; 